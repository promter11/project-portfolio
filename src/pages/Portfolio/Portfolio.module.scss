@import "../../styles/_variables";
@import "../../styles/_extends";

.portfolio {
  height: 100%;
  padding: 8rem 14rem;
  background: rgba(0, 0, 0, 0.75) url("../../assets/images/Portfolio.jpg")
    no-repeat center center / cover;
  background-blend-mode: darken;

  .title {
    @extend %section-main-title;
  }
}
