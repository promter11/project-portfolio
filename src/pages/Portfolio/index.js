import React from "react";

import Heading from "../../components/Heading";
import Container from "../../components/Container";
import Carousel from "../../components/Carousel";

import { portfolioText } from "../../common/jsonText/portfolioText";

import styles from "./Portfolio.module.scss";

const Portfolio = ({ language }) => {
  return (
    <section className={styles.portfolio}>
      <Container>
        <Heading className={styles.title} level={1}>
          {portfolioText[language].title}
        </Heading>
        <Carousel slides={portfolioText[language].content} />
      </Container>
    </section>
  );
};

export default Portfolio;
