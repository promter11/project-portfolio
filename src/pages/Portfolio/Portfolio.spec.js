import React from "react";
import { shallow } from "enzyme";

import Portfolio from "./index";

const props = {
  language: "RU",
};

describe("Portfolio page", () => {
  it("renders correctly with props", () => {
    const wrapper = shallow(<Portfolio {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
