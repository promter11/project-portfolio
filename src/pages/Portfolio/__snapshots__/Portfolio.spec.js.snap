// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio page renders correctly with props 1`] = `
<section
  className="portfolio"
>
  <Container>
    <Heading
      className="title"
      level={1}
    >
      Работы
    </Heading>
    <Carousel
      slides={
        Array [
          Object {
            "date": "24 декабря, 2020",
            "description": "Приложение для заказа парфюмерии",
            "fullDescription": "Приложение для заказа парфюмерии с помощью React + MobX с фильтрами, поиском, корзиной с оформлением заказа.",
            "image": "Portfolio Item 1.jpg",
            "link": Object {
              "path": "/",
              "text": "Подробнее",
            },
            "technologies": Array [
              "HTML (JSX)",
              "Styled Components",
              "React JS (Class Components)",
              "MobX",
              "React Router",
              "Webpack",
            ],
            "title": "Olivia",
          },
          Object {
            "date": "12 августа, 2020",
            "description": "SPA-приложение для скачивания обоев для рабочего стола",
            "fullDescription": "SPA-приложение для загрузки обоев для рабочего стола с использованием библиотеки React на функциональных компонентах и хуках, с обращением к Pixabay API.",
            "image": "Portfolio Item 2.jpg",
            "link": Object {
              "path": "/",
              "text": "Подробнее",
            },
            "technologies": Array [
              "HTML (JSX)",
              "CSS Modules",
              "Typescript",
              "React (Hooks)",
              "Redux + Thunk",
              "React Router",
              "Webpack",
            ],
            "title": "Passion",
          },
          Object {
            "date": "06 мая, 2020",
            "description": "Рекламный лендинг о мобильном приложении AppLab",
            "fullDescription": "Рекламный Landing Page про мобильное приложение AppLab с простыми анимациями, bootstrap сеткой/компонентами и небольшой функциональностью на jQuery.",
            "image": "Portfolio Item 3.jpg",
            "link": Object {
              "path": "/",
              "text": "Подробнее",
            },
            "technologies": Array [
              "HTML5",
              "SCSS",
              "jQuery",
              "Bootstrap 4",
              "Swiper Slider",
              "AOS JS (Animations)",
              "Gulp",
            ],
            "title": "AppLab",
          },
          Object {
            "date": "14 апреля, 2020",
            "description": "Адаптивный Landing Page на тему COVID-19",
            "fullDescription": "Адаптивный Landing Page на тему COVID-19 с небольшой функциональностью в виде плавных переходов, слайдера, модального окна и т.д., а также актуальными данными о количестве зараженных по странам, полученные по обращению к ninja-api через fetch.",
            "image": "Portfolio Item 4.jpg",
            "link": Object {
              "path": "/",
              "text": "Подробнее",
            },
            "technologies": Array [
              "PUG",
              "SCSS",
              "Vanilla JS",
              "Gulp",
            ],
            "title": "COVID-19",
          },
        ]
      }
    />
  </Container>
</section>
`;
